routes/dms.js

```diff
@@ -132,57 +195,20 @@ module.exports = function () {
 
     // Create a visualization per resource as needed
     datapackage.resources.forEach((resource, index) => {
-      // Convert bytes into human-readable format:
-      resource.size = resource.size ? bytes(resource.size, {decimalPlaces: 0}) : resource.size
       
-      let controls = {
-        showChartBuilder: false,
-        showMapBuilder: false
-      }
-
-      const view = {
-        id: index,
-        title: resource.title || resource.name,
-        resources: [
-           resource.name
-        ],
-        specType: null
-      }
-
-      resource.format = resource.format.toLowerCase()
+      // Create share links
+      const shareUrl = `${config.get('SITE_URL')}/${req.params.owner}/${req.params.name}/${resource.id}`
```

Comment:
I think that this speaks to some ambiguity in the existing datapackage spec. I find some thing about it confusing and counter-intuitive. Maybe I just need to study up. I do think some of the choices were not the best. Not sure if is too late to address that here.

1. An optional `id` and a required `name` is confusing. I think it would be more in line with common data practices (db for instance) to require a unique id (something more likely to be unique than a user-provided name, such as RFC-4112 uuid or similar. Not sure how that decision is made or if it can be changed?

2. I don't like the idea of using `index` value as an identifier outside of the context it was generated in -- an index is a convenience for rendering, not an attribute of the view, resource, etc. For instance, if the dataset loses or gains a resource and the index value changes, or the order of views / resources is changed, a unique id will still be valid, while the index will be invalidated.

2a. Same thing if the name changes. /name will be invalid but /uuid will be valid

3. As I understand it, for the dms routes we are enforcing a single view per resource, and we are generating them deterministically - ie, a resource will always get the same view, based on this algorithm. So, at the iframe view we are saying: generate the default view for this resource and display it on a blank page, so we can render it with the iframe.

Reply:
And more generally -  a datapackage defines a list of resources. Adding views here is confusing and mixes presentation with description, and I think has led to a lot of confusion, for me at least.

Reply:
@starsinmypockets `name` of a resource is a unique identifier - you cannot have 2 resources with same name within a datapackage.

Reply:
It would be `name` or `index`. The index is useful when you just playing around I think and if you're going to put an iframed page into your production site then you should use name.

Reply:
Generally, I'm just sharing our experience from datahub.io